api/spec/controllers/spree/api/address_book_controller_spec.rb

```diff
@@ -0,0 +1,60 @@
+require 'spec_helper'
+
+module Spree
+  describe Api::AddressBookController, :type => :controller do
+    render_views
+
+    context "unauthorized user" do
+      it "get 401 on /mine" do
+        api_get :mine
+        expect(response.status).to eq 401
+      end
+
+      it "get 401 on /update" do
+        api_post :update
+        expect(response.status).to eq 401
+      end
+
+      it "get 401 on /remove" do
+        api_post :remove, address_id: 1
+        expect(response.status).to eq 401
+      end
+    end
+
+    context "authorized user with addresses" do
+      let(:address1) { create(:address) }
+      let(:address2) { address1.dup.tap {|a| a.update!(firstname: "Different")} }
+
+      before do
+        stub_authentication!
+        current_api_user.save_in_address_book(address1.attributes, true)
+        current_api_user.save_in_address_book(address2.attributes, false)
+      end
+
+      it "gets their address book" do
+        api_get :mine
+        expect(json_response.length).to eq 2
+      end
+
+      it "the first one is default" do
+        api_get :mine
+        first, second = *json_response
```

Comment:
I don't see a reason that they would guaranteed to be returned in this order (unless I missed a default scope somewhere). Maybe it would be better to test that just one of the two was default